teaching_reproducibility.Rmd

```diff
@@ -436,25 +556,43 @@ To incentivize doing this using Git
 (as opposed to the GitHub web user interface)
 we also assign some marks of each assignment to a mechanics grade. 
 For this we assess whether they have at least three
-commits associated with every single assignment.
+commits associated with every single assignment
+and have written meaningful commit messages.
 We do this because we're trying to build these good habits and practices.
 There are good reasons why you use version control, not just to submit your
 homework but to active as a backup, for collaboration, or in case you
 want to go back in time and change things.
 By the end of this program the students have version controlled their
-work in over 80 different GitHub repositories. 
-They become very practiced and very used to using version control.
+work in over 80 different GitHub repositories.
 We hope this results in using version control becoming a habit, 
-to the point that if they're going to work on a project it's going to go under
-version control - 
-even when they leave the program and are no longer receiving grades for doing this.
-
-In the UBC Master of Data Science program, we use a homegrown tool 
-([rhomboid](https://github.com/mgelbart/rhomboid)),
-to do this, but there are many tools that one could use to do this, 
-including [GitHub Classroom](https://classroom.github.com/) and 
-[Classy](https://github.com/ubccpsc/classy).
-
+to the point that if they're going to work on a project
+it's going to go under version control by default - 
+even when they leave the program
+and are no longer receiving grades for doing this.
+
+One exciting technology that we have recently started incorporating
+in our teaching of reproducible workflows is GitHub actions.
+This tool has allowed us
+to automate the building of individual "playgrounds" of complex Git scenarios
+that would take much effort and typically fail to stage in a large classroom.
+One of the GitHub repositories that we created for these activities
+(https://github.com/ttimbers/review-my-pull-request)
+serves the purpose of providing a playground
+where students can explore and practice
+to learn how to use GitHub's code review feature for pull requests.
+To use it,
+students create their own copy of the repository on GitHub,
+create a branch named pr and then a pull request is automatically created for them by a bot.
+After this simple setup,
+the students can spend the rest of the exercise
+exploring how to perform code reviews on GitHub.
+
+<!--<!-1-TODO I don't really know where this fits in and how we use rhmoboid? -1-> -->
+<!--In the UBC Master of Data Science program, we use a homegrown tool --> 
+<!--([rhomboid](https://github.com/mgelbart/rhomboid)), -->
+<!--to do this, but there are many tools that one could use to do this, --> 
+<!--including [GitHub Classroom](https://classroom.github.com/) and --> 
+<!--[Classy](https://github.com/ubccpsc/classy). -->
```

Comment:
I don't really know how this fits in and how we use rhomboid

Reply:
Rhomboid is GitHubLMS to the world outside MDS. I think it is useful to point instructors who would want to do what we do, places to get started.